src/decls.js

```diff
@@ -33,9 +33,7 @@ export default function bemReactCore(options, BaseComponent, classNameBuilder) {
                         base,
                         fields,
                         {
-                            displayName : classNameBuilder.stringify(
-                                fields.block, fields.elem
-                            ),
+                            displayName : `b:${fields.block}${fields.elem ? ` e:${fields.elem}` : '' }`,
```

Comment:
@veged I can suggest auto conversion to PascalCase only. Because naming can be different, but here we don't know what delimiters we need to use.